mmcls/core/evaluation/mean_ap.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import torch
+
+
+def average_precision(pred, target):
+    """ Calculate the average precision for a single class
+
+    Args:
+        pred (np.ndarray): The model prediction with shape (N, ).
+        target (np.ndarray): The target of each prediction with shape (N, ).
+
```

Comment:
it would help to better understand by writing evaluation formula in docstring.
 
FYI. https://github.com/scikit-learn/scikit-learn/blob/42aff4e2e/sklearn/metrics/_ranking.py#L118
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html?highlight=average_precision#sklearn.metrics.average_precision_score